### PR TITLE
Rename RepartitionExec metric `repart_time` to `repartition_time`

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Closes https://github.com/apache/datafusion/issues/11702

## Rationale for this change

As described on https://github.com/apache/datafusion/issues/11702, this naming is more consistent with other metrics. Thank you @findepi for pointing it out

## What changes are included in this PR?

Rename RepartitionExec metric `repart_time` to `repartition_time`

## Are these changes tested?

CI

## Are there any user-facing changes?
The metric name is now different

### CODE REVIEW

**Review by findepi — APPROVED** [review]

**Review by jonahgao — APPROVED** [review]
Looks good to me!

**Comment by alamb** [comment]
Thanks @jonahgao  and @findepi 

I plan to leave this open for a few days to collect any other comments that people might have

cc @korowa  in case you have any thoughts you would like to share

File: datafusion/physical-plan/src/repartition/mod.rs [hunk 1]
```diff
@@ -415,7 +415,7 @@ struct RepartitionMetrics {
     /// Time in nanos to execute child operator and fetch batches
     fetch_time: metrics::Time,
     /// Time in nanos to perform repartitioning
```
Comment (comphead) [inline datafusion/physical-plan/src/repartition/mod.rs:417]:
```suggestion
    /// Repartitioning elapsed time in nanos
    
```

**Comment by alamb** [comment]
It seems people like this naming better, so I will merge it in. Thanks everyone for your comments and review

**Pull request merged**